compiler/rustc_hir_typeck/src/op.rs

```diff
@@ -148,10 +148,11 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                         rhs_ty,
                         op,
                     );
-                    self.demand_suptype(expr.span, builtin_return_ty, return_ty);
+                    self.demand_eqtype(expr.span, builtin_return_ty, return_ty);
```

Comment:
This `demand_suptype` doesn't matter because the only types we're gonna encounter here are scalars.